library/core/src/ptr/mod.rs

```diff
@@ -1989,9 +1989,18 @@ impl<F: FnPtr> fmt::Debug for F {
 /// as all other references. This macro can create a raw pointer *without* creating
 /// a reference first.
 ///
-/// Note, however, that the `expr` in `addr_of!(expr)` is still subject to all
-/// the usual rules. In particular, `addr_of!(*ptr::null())` is Undefined
-/// Behavior because it dereferences a null pointer.
+/// The `expr` in `addr_of!(expr)` is evaluated as a place expression, but never loads
+/// from the place or requires the place to be dereferenceable. This means that
+/// `addr_of!(*ptr)` is defined behavior even if `ptr` is dangling or misaligned.
```

Comment:
Now it seems like we've forgotten `null`, but this is allowed too, right?
```suggestion
/// `addr_of!(*ptr)` is defined behavior even if `ptr` is null, dangling, or misaligned.
```

Reply:
Sure, I was thinking "null" is just a special case of "dangling", but we can also list it explicitly.